hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/api/TokenServiceApiImpl.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl.api;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.ContractID;
+import com.hedera.hapi.node.contract.ContractNonceInfo;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.node.app.service.token.api.TokenServiceApi;
+import com.hedera.node.app.service.token.impl.WritableAccountStore;
+import com.hedera.node.app.spi.store.WritableStoreFactory;
+import com.swirlds.config.api.Configuration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+
+/**
+ * Implements {@link TokenServiceApi} via {@link WritableAccountStore} calls.
+ */
+public class TokenServiceApiImpl implements TokenServiceApi {
+    private final WritableStoreFactory storeFactory;
+
+    public TokenServiceApiImpl(@NonNull final Configuration config, @NonNull final WritableStoreFactory storeFactory) {
+        requireNonNull(config);
+        this.storeFactory = requireNonNull(storeFactory);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void createAndMaybeAliasContract(
+            @NonNull final ContractID idToCreate, @NonNull final Consumer<Account.Builder> spec) {
+        requireNonNull(spec);
+        requireNonNull(idToCreate);
+        final var builder = Account.newBuilder()
+                .accountId(AccountID.newBuilder().accountNum(idToCreate.contractNumOrThrow()))
+                .smartContract(true);
+        spec.accept(builder);
+        final var newContract = builder.build();
+        if (newContract.tinybarBalance() > 0) {
```

Comment:
Do accounts have more invariants? If they do, we should check all of them here. It is the responsibility of the TokenService to ensure invariants, and we should not rely on the called code to do the right thing.